src/main/java/game/GameFlow.java

```diff
@@ -8,25 +8,32 @@
 public class GameFlow {
 
     private final Ui ui;
-    private final HumanPlayer humanPlayer;
+    private HumanPlayer humanPlayer;
     private final Computer computer;
 
-    public GameFlow(Ui ui, HumanPlayer humanPlayer, Computer computer) {
+    public GameFlow(Ui ui, Computer computer) {
         this.ui = ui;
-        ui.welcomePlayer();
-        this.humanPlayer = humanPlayer;
         this.computer = computer;
+        gameSetUp();
     }
 
     public void run() {
-        Move userMove = humanPlayer.makeMove();
+        Move userMove = humanPlayer.makeMove(ui);
+
+        Move computerMove = computer.makeMove(ui);
 
-        Move computerMove = computer.makeMove();
-        ui.declareComputerMove(computerMove.getName());
+        printResult(userMove.playAgainst(computerMove));
 
-        String verdict = userMove.playAgainst(computerMove);
+        if (ui.playAgain().equals("yes")) {
+            gameSetUp();
```

Comment:
Maybe a bit strange from a user experience point of view, if you're asked to play again or not, that you have to start right from the beginning again, i.e. have to choose the language again and to be greeted again. I would assume, once a user chose a language, they will play in that language.

Reply:
True. I was waiting for your opinion about this (I should have directly asked, maybe)